improver/cli/__init__.py

```diff
@@ -239,27 +242,29 @@ def constrained_inputcubelist_converter(to_convert):
                 than one match is found.
         """
         from improver.utilities.load import load_cube
+        from iris import load_raw
         from iris.cube import CubeList
-
-        cubelist = CubeList()
-        for constr in constraints:
-            constr_list = [constr] if isinstance(constr, str) else constr
-            found_cubes = []
-            for constr_item in constr_list:
-                try:
-                    found_cubes.append(maybe_coerce_with(
-                        load_cube, to_convert, constraints=constr_item))
-                except ValueError:
-                    pass
-            if len(found_cubes) != 1:
-                msg = (f"Incorrect number of valid inputs available for the "
-                       "{constr} constraint. "
-                       f"Number of valid inputs: {len(found_cubes)} "
-                       f"The valid inputs found are: {found_cubes}")
-                raise ValueError(msg)
-            cubelist.extend(found_cubes)
-
-        return cubelist
+        # Iris will add on an extra cube which isn't to be counted.
```

Comment:
Not exactly.  How about:
```
# File includes metadata prefix cube which should not be counted
```
But I'm wondering if this now constrains us only to be able to read files saved from IMPROVER.  We do need this code to be able to handle files which do not include a prefix cube.

Reply:
I wrongly understood there to always be a metadata cube, due to this, the options are:
1. write an extra function that checks if there is a metadata cube.
2.  Carry on blindly if the number of constraints and cubs mismatch